### PR TITLE
Upgrade dependencies and add mocha testing to dev

### PR DESCRIPTION
**Pull request by markschaake**
Upgrades all dependencies, sets up mocha as a binary, and provides example testing in dev project.

@codeviking please review

### CODE REVIEW

File: dev/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": [ "react", "es2015" ]
```
Comment (markschaake) [inline dev/.babelrc]: this is needed for mocha to work with react and ES6 sources

File: dev/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "repository": "allenai/syrup",
   "license": "Apache-2.0",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "mocha --compilers js:babel-register **/*_test.js",
```
Comment (markschaake) [inline dev/package.json]: ugly, but this is what you have to add to a project to enable mocha testing with all the babel goodies

File: dev/app/util_test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/*eslint-env node, mocha */
```
Comment (codeviking) [inline dev/app/util_test.js]: Is this for eslint?

File: dev/gulpfile.js [hunk 4]
```diff
@@ -2,7 +2,10 @@
 
 var gulp = require('gulp');
 var gutil = require('gulp-util');
-var syrup = require('../');
+// this requires you have syrup available globally. Dev workflow is to generate a link
+// locally to syrup and then dev on this dev project. To create the syrup link:
+// $ cd path/to/syrup && npm link
```
Comment (codeviking) [inline dev/gulpfile.js]: Thanks for the doc!

File: dev/package.json (hunk 2 above)
Comment (codeviking) [inline dev/package.json]: Makes sense, thanks. :+1:

File: tests/gulp-test.js [hunk 5]
```diff
@@ -110,9 +108,9 @@ describe('syrup.gulp.init()', function() {
     gulp.start('less', function() {
       assert(fs.existsSync(path.resolve(PROJECT_BUILD_DIR, 'foo.css')));
       assert.equal(
-          fs.readFileSync(path.resolve(PROJECT_BUILD_DIR, 'foo.css')).toString(),
-          fs.readFileSync(path.resolve(PROJECT_ROOT, 'foo.expected.css')).toString()
-        );
+        fs.readFileSync(path.resolve(PROJECT_BUILD_DIR, 'foo.css')).toString().trim(),
+        fs.readFileSync(path.resolve(PROJECT_ROOT, 'foo.expected.css')).toString().trim()
```
Comment (codeviking) [inline tests/gulp-test.js]: :+1:

**Comment by codeviking** [comment]
LGTM!  I'll merge this tomorrow and cut a new syrup version, or would you prefer to do that (I can add you so that you can publish releases if you'd like).

**Pull request merged**